src/click/core.py

```diff
@@ -1224,7 +1224,7 @@ def format_help_text(self, ctx, formatter):
             with formatter.indentation():
                 help_text = self.help
                 if self.deprecated:
-                    help_text += DEPRECATED_HELP_NOTICE
+                    help_text = DEPRECATED_HELP_NOTICE + "\n" + help_text
```

Comment:
After thinking about the formatting, I think putting the notice at the front of the line instead of on its own line looks better. Also, the current notice has a space at the front, which can be removed since it's being prepended instead of appended now.